test/videojs.ads.test.js

```diff
@@ -1,1025 +1,732 @@
+  assert.strictEqual(this.player.ads.state, 'content-playback');
+});
+
+QUnit.test('adtimeout in postroll? transitions to content-playback and fires ended', function(assert) {
+  assert.expect(8);
+  assert.strictEqual(this.player.ads.state, 'content-set');
+  this.player.trigger('adsready');
+  assert.strictEqual(this.player.ads.state, 'ads-ready');
+  this.player.tech.trigger('play');
+  this.player.trigger('adtimeout');
+  this.player.trigger('ended');
+  assert.strictEqual(this.player.ads.state, 'postroll?');
+  this.player.ads.snapshot.ended = true;
+  this.player.trigger('adtimeout');
+  assert.strictEqual(this.player.ads.state, 'content-resuming');
+  assert.strictEqual(this.player.ads.triggerevent, 'adtimeout', 'adtimeout should be the trigger event');
+  this.clock.tick(1);
+  assert.strictEqual(this.contentPlaybackSpy.callCount, 2, 'A content-playback event should have been triggered');
+  assert.strictEqual(this.contentPlaybackReason(1), 'ended', 'The reason for content-playback should have been ended');
+  assert.strictEqual(this.player.ads.state, 'content-playback');
+});
+
+QUnit.test('adskip in postroll? transitions to content-playback and fires ended', function(assert) {
+  assert.expect(8);
+  assert.strictEqual(this.player.ads.state, 'content-set');
+  this.player.trigger('adsready');
+  assert.strictEqual(this.player.ads.state, 'ads-ready');
+  this.player.tech.trigger('play');
+  this.player.trigger('adtimeout');
+  this.player.trigger('ended');
+  assert.strictEqual(this.player.ads.state, 'postroll?');
+  this.player.ads.snapshot.ended = true;
+  this.player.trigger('adskip');
+  assert.strictEqual(this.player.ads.state, 'content-resuming');
+  assert.strictEqual(this.player.ads.triggerevent, 'adskip', 'adskip should be the trigger event');
+  this.clock.tick(1);
+  assert.strictEqual(this.contentPlaybackSpy.callCount, 2, 'A content-playback event should have been triggered');
+  assert.strictEqual(this.contentPlaybackReason(1), 'ended', 'The reason for content-playback should have been ended');
+  assert.strictEqual(this.player.ads.state, 'content-playback');
+});
+
+QUnit.test('an ended event is fired in content-resuming via a timeout if not fired naturally', function(assert) {
+  assert.expect(6);
+  var spy = sinon.spy();
+  this.player.on('ended', spy);
+  assert.strictEqual(this.player.ads.state, 'content-set');
+  this.player.trigger('adsready');
+  assert.strictEqual(this.player.ads.state, 'ads-ready');
+  this.player.tech.trigger('play');
+  this.player.trigger('adtimeout');
+  this.player.trigger('ended');
+  assert.strictEqual(this.player.ads.state, 'postroll?');
+  this.player.ads.startLinearAdMode();
+  this.player.ads.snapshot.ended = true;
+  this.player.ads.endLinearAdMode();
+  assert.strictEqual(this.player.ads.state, 'content-resuming');
+  assert.strictEqual(spy.callCount, 0, 'we should not have gotten an ended event yet');
+  this.clock.tick(2000);
```

Comment:
I'll change this and other timeouts. It makes sense to me that they should only be ticking enough to trigger the behavior - otherwise, it might hide weird edge-cases or something.